tests/admin/controller/test_controller.py

```diff
@@ -1752,14 +1752,21 @@ def test_bulk_circulation_events(self):
         get_one_or_create(self._db, WorkGenre, work=self.english_1, genre=genres[0], affinity=0.2)
 
         time = datetime.now() - timedelta(minutes=1)
-        get_one_or_create(
+        event, ignore = get_one_or_create(
             self._db, CirculationEvent,
-            license_pool=lp, type=CirculationEvent.DISTRIBUTOR_CHECKOUT, start=time, end=time)
+            license_pool=lp, type=CirculationEvent.DISTRIBUTOR_CHECKOUT,
+            start=time, end=time
+        )
         time += timedelta(minutes=1)
 
+        # Try an end-to-end test, getting all circulation events for
+        # the current day.
         with self.app.test_request_context("/"):
             response, requested_date, date_end, library_short_name = self.manager.admin_dashboard_controller.bulk_circulation_events()
-        reader = csv.reader([row for row in response.split("\r\n") if row], dialect=csv.excel)
+        reader = csv.reader(
+            [row for row in response.split("\r\n") if row],
+            dialect=csv.excel
+        )
```

Comment:
These changes don't matter -- I just did them to make the code easier for me to read and test.